admin-cli/src/index.ts

```diff
@@ -121,32 +120,76 @@ namespace v3v4 {
     }
   }
 
-  // "Auto-" accept upgrade proposals.
-  export const accept = async (ledgerParams: LedgerParams, company: string) => {
+  export const upgradeAccept = async (ledgerParams: LedgerParams, company: string) => {
     const companyLedger = connect(ledgerParams, company);
-    const employees = await companyLedger.query(davl3.EmployeeRole, {});
-    for (const employeeRole of employees) {
-      const employee = employeeRole.key;
+    const employeeRoleContracts = await companyLedger.query(davl3.EmployeeRole, {});
+    // Create update agreements.
+    for (const employeeRoleContract of employeeRoleContracts) {
+      const employee = employeeRoleContract.key;
       const employeeLedger = connect(ledgerParams, employee);
-      const [upgradeProposal] =
-        await employeeLedger.query(davlUpgradev3v4.UpgradeProposal, { employee: employee });
-      await employeeLedger.exercise(
+      const [upgradeProposalContract] =
+        await employeeLedger.query(davlUpgradev3v4.UpgradeProposal, { employee });
+        await employeeLedger.exercise(
           davlUpgradev3v4.UpgradeProposal.UpgradeProposal_Accept,
-          upgradeProposal.contractId,
+          upgradeProposalContract.contractId,
           {},
         );
       console.log(`Accepted UpgradeProposal for ${employee}.`);
     }
   }
+
+  export const upgradeFinish = async (ledgerParams: LedgerParams, company: string) => {
+    const companyLedger = connect(ledgerParams, company);
+    const employeeUpgradeAgreementContracts =
+      await companyLedger.query(davlUpgradev3v4.UpgradeAgreement, { company });
+    for (const employeeUpgradeAgreementContract of employeeUpgradeAgreementContracts) {
+      const employee = employeeUpgradeAgreementContract.payload.employee;
+      const employeeRoleContract = await companyLedger.lookupByKey(davl4.EmployeeRole, employee);
+      if (employeeRoleContract) {
+        const employeeRole = employeeRoleContract.payload;
+        const vacationRequestContracts = await companyLedger.query(davl3.VacationRequest, { vacation: { employeeRole } });
+        if (vacationRequestContracts.length) {
+          for (const vacationRequestContract of vacationRequestContracts) {
+            const fromDate = vacationRequestContract.payload.vacation.fromDate;
+            const toDate = vacationRequestContract.payload.vacation.toDate;
```

Comment:
```suggestion
            const {toDate, fromDate} = vacationRequestContract.payload.vacation;
```
and remove the lone above.